src/components/feedback/Dialog.tsx

```diff
@@ -108,31 +117,37 @@ const DialogNext = function Dialog({
     [dialogDescriptionId, modalRef]
   );
 
+  // Wrap modal dialogs in a full-screen overlay
+  const Wrapper = modal ? Overlay : Fragment;
+
   return (
-    <Overlay>
+    <Wrapper>
       <div
+        aria-modal={modal}
         data-component="Dialog"
         tabIndex={-1}
         // NB: Role can be overridden with an HTML attribute; this is purposeful
         role="dialog"
         {...htmlAttributes}
         className={classnames(
-          // Maximum width and height should not exceed 90vw/h
-          'max-w-[90vw] max-h-[90vh]',
-          // Overlay sets up a flex layout centered on both axes. For taller
-          // viewports, remove this modal container from the flex flow with
-          // fixed positioning and position it 10vh from the top of the
-          // viewport. This feels more balanced on taller screens. Ensure an
-          // equal 10vh gap at the bottom of the screen by adjusting max-height
-          // to `80vh`.
-          'tall:fixed tall:max-h-[80vh] tall:top-[10vh]',
           // Column-flex layout to constrain content to max-height
           'flex flex-col',
+          {
+            // Maximum width and height should not exceed 90vw/h
+            'max-w-[90vw] max-h-[90vh]': modal,
+            // Overlay sets up a flex layout centered on both axes. For taller
+            // viewports, remove this modal container from the flex flow with
+            // fixed positioning and position it 10vh from the top of the
+            // viewport. This feels more balanced on taller screens. Ensure an
+            // equal 10vh gap at the bottom of the screen by adjusting max-height
+            // to `80vh`.
+            'tall:fixed tall:max-h-[80vh] tall:top-[10vh]': modal,
+          },
           {
             // Max-width rules will ensure actual width never exceeds 90vw
-            'w-[30rem]': width === 'sm',
-            'w-[36rem]': width === 'md', // default
-            'w-[42rem]': width === 'lg',
+            'w-[30rem]': modal && width === 'sm',
+            'w-[36rem]': modal && width === 'md', // default
+            'w-[42rem]': modal && width === 'lg',
```

Comment:
This application of sizing styles will be refined as implementation continues, with the introduction of some of the customization controls presented last week. But for now: Only assert size and positioning for modal Dialogs.